e2e-tests/server.py

```diff
@@ -0,0 +1,36 @@
+import http.server
+import os
+import ssl
+
+def remove_file(path):
+    try:
+        os.remove(path)
+    except:
+        pass
+
+remove_file('ca.pem')
+remove_file('ca.srl')
+remove_file('ca.key')
+remove_file('server.csr')
+remove_file('server.ext')
+remove_file('server.pem')
+remove_file('server.key')
+os.system('openssl req -newkey rsa:4096 -x509 -sha256 -days 1 -nodes -out ca.pem -keyout ca.key -subj "/C=UT/CN=ca.localhost"')
+os.system('openssl req -new -newkey rsa:4096 -sha256 -nodes -out server.csr -keyout server.key -subj "/C=UT/CN=localhost"')
+with open('server.ext', 'w') as file:
+    file.write('''
+authorityKeyIdentifier=keyid,issuer
+basicConstraints=CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = localhost
+''')
+os.system('openssl x509 -req -in server.csr -CA ca.pem -CAkey ca.key -CAcreateserial -out server.pem -days 1 -sha256 -extfile server.ext')
```

Comment:
CI failed supuriously since this cmd is executed after cargo-binstall tries to load the root certificate.

I think it's better to move them into signing.sh to make sure they are executed before cargo-binstall.

Reply:
The CI failure is quite consistent, looks like openssl key generation is much slower than the 5s sleep

Reply:
eh, the ci failing only on mac and windows leads me to think that no, it's not delay based, it's just broken on mac and windows

Reply:
On windows it seems like the `signing-test` cannot be executed (is that an elf?) https://github.com/cargo-bins/cargo-binstall/actions/runs/6095627429/job/16639287687?pr=1345#step:5:980

On MacOS, I can't see the output from the python server because its output is directed to null.

I still think it's better to move the keygen into the shell script and keep the python script as simple as possible.
Ideally I'd like no python file at all, so that I don't have to look at two different files to understand what it is doing, but as long as the CI succeeds I'm ok to merge this.